keyboards/planck/keymaps/floodedcodeboy/keymap.c

```diff
@@ -0,0 +1,341 @@
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_planck_grid(
+    _______, QK_BOOT, DB_TOGG, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_DEL ,
+    _______, _______, MU_NEXT, AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  _______, RGB_M_B, RGB_M_SW, RGB_M_R,
+    _______, AU_PREV, AU_NEXT, MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, RGB_SPI, RGB_SPD, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
+
+// COLOR LAYERS
+const rgblight_segment_t PROGMEM rgb_layer_base[] = RGBLIGHT_LAYER_SEGMENTS(
+    {3, 1, HSV_GREEN}
+);
+
+const rgblight_segment_t PROGMEM rgb_layer_raise[] = RGBLIGHT_LAYER_SEGMENTS(
+    {5, 1, HSV_CYAN}
+);
+
+const rgblight_segment_t PROGMEM rgb_layer_lower[] = RGBLIGHT_LAYER_SEGMENTS(
+    {4, 1, HSV_YELLOW}
+);
+
+const rgblight_segment_t PROGMEM rgb_layer_adj[] = RGBLIGHT_LAYER_SEGMENTS(
+    {3, 1, HSV_PURPLE}
+);
+
+const rgblight_segment_t PROGMEM rgb_layer_shift[] = RGBLIGHT_LAYER_SEGMENTS(
+    {6, 1, HSV_ORANGE}
+);
+
+// MOD KEY COLOUR LAYERS
+const rgblight_segment_t PROGMEM rbg_layer_mod1[] = RGBLIGHT_LAYER_SEGMENTS(
+    {8, 1, HSV_MAGENTA}
+);
+const rgblight_segment_t PROGMEM rgb_layer_mod2[] = RGBLIGHT_LAYER_SEGMENTS(
+    {1, 1, HSV_PURPLE}
+);
+const rgblight_segment_t PROGMEM rgb_layer_mod3[] = RGBLIGHT_LAYER_SEGMENTS(
+    {2, 1, HSV_AZURE}
+);
+
+const rgblight_segment_t* const PROGMEM _rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    rgb_layer_base,
+    rgb_layer_raise,    // Overrides caps lock layer
+    rgb_layer_lower,    // Overrides other layers
+    rgb_layer_adj,     // Overrides other layers
+    rgb_layer_shift,
+    rbg_layer_mod1,
+    rgb_layer_mod2,
+    rgb_layer_mod3
+);
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = _rgb_layers;
+}
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+// layer_state_t default_layer_state_set_user(layer_state_t state) {
+//     return state;
+// }
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case LOWER:
+      if (record->event.pressed) {
+        rgblight_set_layer_state(1, true);
+      } else {
+        rgblight_set_layer_state(1, false);
+      }
```

Comment:
you could abbreviate this with:
```suggestion
      rgblight_set_layer_state(1, record->event.pressed);
```

However, this would be much better in the `layer_state_set_user` function, as then you can use any keycode for the layers.